File: Programmers/Level1/체육복/체육복.js
function solution(n, lost, reserve) {
    let _lost = lost.filter((v) => !reserve.includes(v));
    let _reserve = reserve.filter((v) => !lost.includes(v));

    for (let i = 0; i < _reserve.length; i++) {
        if (_lost.includes(_reserve[i] - 1)) {
            _lost = _lost.filter((v) => v !== _reserve[i] - 1);
        } else if (reserve.includes(lost[i] + 1)) {
            _lost = _lost.filter((v) => v !== _reserve[i] + 1);
        }
    }
    return n - _lost.length;
}

File: Programmers/Level1/체육복/체육복.py
def solution(n, lost, reserve):
    reserve_student = set(reserve) - set(lost)
    lost_student = set(lost) - set(reserve)
    
    for rs in reserve_student:
        if rs - 1 in lost_student:
            lost_student.remove(rs - 1)
        elif rs + 1 in lost_student:
            lost_student.remove(rs + 1)
            
    return n - len(lost_student)